tests/unit/jobserver/test_middleware.py

```diff
@@ -137,6 +137,21 @@ class DummyView(DetailView):
     assert response.context_data["template_name"] == "my_template"
 
 
+def test_template_name_middleware_with_template_name_as_a_list_without_a_name(rf):
+    project = ProjectFactory()
+
+    class DummyView(DetailView):
+        model = Project
+        template_name = ""
+
+    request = rf.get("/")
+    response = DummyView.as_view()(request, pk=project.pk)
+
+    TemplateNameMiddleware(None).process_template_response(request, response)
+
+    assert response.context_data["template_name"] is None
```

Comment:
thought: This seems OK to capture the existing behaviour, but I'm not sure if this state can be reached, and if so, does it make sense to allow it.

That is: I don't know enough about how this `template_name` is created and how it might interact elsewhere. Would it be better to raise an exception here?

Reply:
@lucyb: Thanks for investigating! I wasn't entirely sure where to start with it.

If I have understood you correctly, I tried switching the test to use a ListView, with `get_queryset()` returning an empty list of objects.

This still resulted in a non-zero length list as `response.template_name`.

So I think your assertion here is correct, that a list is never empty. I'll amend the code accordingly.